### PR TITLE
Documentation site

### PR DESCRIPTION
**Pull request by fahad19**
## What's done

* Documentation site generator
* All content written in Markdown like before
* A fresh new design

## Usage

All content placed under `./site` directory.

```
$ npm install

$ make site-serve
Visit http://localhost:6001
```

The generated site will be available in `./_site`, which can later be pushed to `gh-pages` branch, and will be made available under https://frint.js.org

## Logo

I am not a designer, so used [Pacifico](https://fonts.google.com/specimen/Pacifico) font (from Google Fonts) for generating a typographic logo.

Anyone who needs the original PSDs, ping me.

## AFTER merge

* Write more docs
* Find a better image for homepage under `Modular & Extensible`
* Find better wordings for homepage.
* The hero buttons need more styling
* Better colours?
* Hero should have a gradient?
* Delete `./docs` directory in another PR later
* Under `./site/assets`, write cleaner JS/CSS

## Preview

screen shot 2017-03-09 at 11 03 27

screen shot 2017-03-09 at 11 03 57

### CODE REVIEW

File: site/assets/css/site.scss [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    font-size: 24px;
+    margin-top: 10px;
+  }
+
+  p {
+    margin-top: 15px;
+  }
+}
+
+// .menu-container.docs {
```
Comment (AlexDudar) [inline site/assets/css/site.scss]: do you need this?

File: site/assets/css/site.scss [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+// }
+
+.menu.docs {
+  .menu-list {
+    li {
+      ul {
+        border-left: 1px solid #eee;
+
+        li a {
+          font-size: 90%;
```
Comment (AlexDudar) [inline site/assets/css/site.scss]: why % ?

File: site/layouts/home.html [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+            <p>
+              We embraced the power of reactive programming everywhere.
+            </p>
+          </div>
+        </div>
+      </div>
+    </section>
+
+    <!--
```
Comment (AlexDudar) [inline site/layouts/home.html]: can be removed

**Comment by fahad19** [comment]
@AlexDudar: it's one of the `AFTER merge` tasks to clean up site's CSS/JS further. is it alright if we do that later? this would enable others to start writing more docs sooner.

Besides, it's not going to `master` yet.

**Pull request merged**